File: app/src/main/java/io/lerk/lrkFM/activities/file/AbstractLoader.java
package io.lerk.lrkFM.activities.file;

import android.support.annotation.Nullable;

import java.util.ArrayList;

import io.lerk.lrkFM.entities.FMFile;
import io.lerk.lrkFM.exceptions.BlockingStuffOnMainThreadException;
import io.lerk.lrkFM.exceptions.EmptyDirectoryException;
import io.lerk.lrkFM.exceptions.NoAccessException;

/**
 * Because I like to rant about <pre>ILoader</pre> 'nsuch.
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
public abstract class AbstractLoader {

    abstract ArrayList<FMFile> loadLocationFiles() throws NoAccessException, EmptyDirectoryException, BlockingStuffOnMainThreadException;

    abstract protected ArrayList<FMFile> loadLocationFilesForPath(@Nullable String parent) throws NoAccessException, EmptyDirectoryException, BlockingStuffOnMainThreadException;


}

File: app/src/main/java/io/lerk/lrkFM/activities/file/AbstractLoader.kt
package io.lerk.lrkFM.activities.file

import io.lerk.lrkFM.entities.FMFile
import io.lerk.lrkFM.exceptions.BlockingStuffOnMainThreadException
import io.lerk.lrkFM.exceptions.EmptyDirectoryException
import io.lerk.lrkFM.exceptions.NoAccessException

/**
 * Because I like to rant about <pre>ILoader</pre> 'nsuch.
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
abstract class AbstractLoader {
    @Throws(
        NoAccessException::class,
        EmptyDirectoryException::class,
        BlockingStuffOnMainThreadException::class
    )
    abstract fun loadLocationFiles(): ArrayList<FMFile>
    @Throws(
        NoAccessException::class,
        EmptyDirectoryException::class,
        BlockingStuffOnMainThreadException::class
    )
    protected abstract fun loadLocationFilesForPath(parent: String?): ArrayList<FMFile>
}
